pkg/sendconfig/sendconfig_test.go

```diff
@@ -128,3 +128,51 @@ func Test_updateReportingUtilities(t *testing.T) {
 	assert.True(t, hasSHAUpdateAlreadyBeenReported([]byte("yet-another-fake-sha")))
 	assert.True(t, hasSHAUpdateAlreadyBeenReported([]byte("yet-another-fake-sha")))
 }
+
```

Comment:
Please add an integration test that will:
- specify a custom list of tags
- verify that entries created in Kong have the tags appropriately set
- verify that preexisting entries not matching the filter tag are not affected

Reply:
@mflendrich, can you explain more about the 3rd point?

Reply:
In DB mode, when KIC is run with a non-empty `--filter-tags`, it causes the following:
- all resources that KIC creates in Kong Admin API have the tag specified
- for sync purposes KIC sees only those resources in the Kong Admin API that have the tag specified

Therefore, resources not matching the tag filter should be "invisible" to KIC and not affected by KIC.

I'm asking for a test that confirms the above behavior. For example:
- KIC and Kong proxy start up
- the test manually creates a route X in Admin API with a "managed-by-ingress-controller" tag
- the test manually creates a route Y in Admin API with a "managed-by-something-else" tag
- KIC tries to sync
- the test verifies that, after sync, X no longer exists in the cluster but Y remains

makes sense?

Reply:
@hbagdi anything to add/comment wrt. tag support behavior?

Reply:
@mflendrich sure will come with the test 👍🏻